base/strings/search.jl

```diff
@@ -164,6 +164,12 @@ end
 
 
 in(c::AbstractChar, s::AbstractString) = (findfirst(isequal(c),s)!==nothing)
+function in(c::Char, s::String)
+    c > '\x7f' && return findfirst(isequal(c),s) !== nothing
```

Comment:
Can't we speed up `findfirst(isequal(c), s)` instead?

Reply:
the problem is that the reason this is slow is that the `in` to `findfirst` to `findnext` to `_search` has repeated bound-checks that add about 8ns of overhead. Fixing the problem here is a lot easier than figuring out which of the internal functions can remove checks without breaking someone's API.

Reply:
Can't we use `@inbounds` and `@propagate_boundscheck` etcetera for this sort of thing?